### PR TITLE
feat: warn on duplicate extension name

### PR DESCRIPTION
**Pull request by Jamalam360**
Very small change but PRing because I know you've said you've been busy, gdude.

### CODE REVIEW

**Comment by Jamalam360** [comment]
I don't _think_ those actions errors are my fault. Builds fine locally, and I can't see why this would cause those errors

**Comment by lukellmann** [comment]
> I don't _think_ those actions errors are my fault. Builds fine locally, and I can't see why this would cause those errors

It's because of Kord's snapshot version in the develop branch, there was a breaking change (the reason for https://github.com/Kord-Extensions/kord-extensions/pull/138)

**Comment by Jamalam360** [comment]
Oh ok that's fine then

**Comment by gdude2002** [comment]
#138 has been merged, so you'll want to rebase

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/ExtensibleBot.kt [hunk 1]
```diff
@@ -266,6 +266,10 @@ public open class ExtensibleBot(
     public open suspend fun addExtension(builder: () -> Extension) {
         val extensionObj = builder.invoke()
 
+        if (extensions.contains(extensionObj.name)) {
+            logger.warn { "Using duplicate extension name: ${extensionObj.name}" }
```
Comment (gdude2002) [inline kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/ExtensibleBot.kt]: This should use `logger.error` instead and unload the old extension first - duplicate extension names are _breaking_, as the already existing extension will no longer be able to be referred to.

**Review by gdude2002 — CHANGES_REQUESTED** [review]

File: kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/ExtensibleBot.kt (hunk 1 above)
Comment (gdude2002) [reply kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/ExtensibleBot.kt]: In case you only look at emails: Edited the original comment

Comment (Jamalam360) [reply kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/ExtensibleBot.kt]: Yeah OK, I wasn't sure if you'd want the default behaviour changing, pushing now

Comment (gdude2002) [reply kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/ExtensibleBot.kt]: The push is good, but the message isn't clear enough. Maybe something like...

`Extension with duplicate name ${extensionObj.name} registered - unloading previously registered extension.`

**Review by gdude2002 — APPROVED** [review]

**Comment by gdude2002** [comment]
Thank!

**Comment by Jamalam360** [comment]
:)

**Pull request merged**